### DISCUSSION TITLE
Game freezing too long caused script termination and black screen

### DISCUSSION THREAD

**Discussion by Michael21107** [scripthookvdotnet/scripthookvdotnet#1367] [General]
I want to fix this script but have no clue how to do so, can u please look at [this issue](https://gitlab.com/Michael21107/RememberLastLocation-GTA5/-/issues/1)?

**Comment by kagikn** [comment]
Try to find out what is the issue and deliver the proper and sufficient information as possible, instead of just asking someone to look at an issue without any clue. I suspect that the script makes the game "stuck" in the black screen because of the poor design though.

**Reply by Michael21107** [reply]
sorry, the problem is i dont have a clue myself and don't know what is sufficient info too

**Reply by kagikn** [reply]
try ask a script dev who are willing to investigate the issue to do then, I don't have infinite time to diagnose every issue that I can't answer within like 10 minutes, especially if there's nearly no effort to narrow the scope of where to investigate.

**Reply by Michael21107** [reply]
understandable, the only problem I found was `[13:18:09] [ERROR] Blocking script! Script LastLocation.Main (file name: LastLocation.dll) was terminated because it caused the game to freeze too long.
[13:18:09] [WARNING] Aborted script LastLocation.Main.
   at System.Threading.Monitor.ObjWait(Boolean exitContext, Int32 millisecondsTimeout, Object obj)
   at System.Threading.SemaphoreSlim.WaitUntilCountOrTimeout(Int32 millisecondsTimeout, UInt32 startTime, CancellationToken cancellationToken)
   at System.Threading.SemaphoreSlim.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at SHVDN.Script.Wait(Int32 ms)
   at LastLocation.Main.Main_Tick(Object sender, EventArgs e)
   at SHVDN.Script.DoTick()
   at SHVDN.Script.MainLoop()`

**Reply by kagikn** [reply]
Glad you have narrowed the scope of the issue a bit, now I can tell you one possible cause of the crash for blocking too long time. I think [ParseWeaponsString](https://gitlab.com/Michael21107/RememberLastLocation-GTA5/-/blob/b0520eab468755706a449d8b132d2a9716f1ef7a/LastLocation/Main.cs#L100) may block long enough (like several seconds) in one tick for SHVDN to stop the script running. May be a flaw, but the script would start parsing the ini because of [the of condition where Game.IsLoading is evaluated](https://gitlab.com/Michael21107/RememberLastLocation-GTA5/-/blob/b0520eab468755706a449d8b132d2a9716f1ef7a/LastLocation/Main.cs#L145). Scripts will load when the loading screen is shown and the loading screen is drawn in a thread other than the main thread where the core game logic and ysc and our scripts run, so you can't tell if the script is blocking other SHVDN scripts from how the loading screen is drawn.

**Reply by Michael21107** [reply]
appreciate it
